### PR TITLE
jump: init at 0.23.0

### PR DESCRIPTION
**Pull request by sondr3**
###### Motivation for this change
[Jump](https://github.com/gsamokovarov/jump) is a Go tool very similar to the well known `z` CLI tool with a few extras.

###### Things done
Installed and ran it on both my NixOS machine and my Mac, checked for shell completion and man pages and that it worked on both.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -4127,6 +4127,8 @@ in
 
   jumanpp = callPackage ../tools/text/jumanpp {};
 
+  jump = callPackage ../tools/system/jump {};
```
Comment (sondr3) [inline pkgs/top-level/all-packages.nix]: I'm not married to where I placed it, so if anyone has a better suggestion I'm all ears.

**Review by tomberek — COMMENTED** [review]
This seems to work without `export HOME` and I don't see a difference in the generated completion scripts.

**Comment by sondr3** [comment]
I couldn't get it to work without it on my NixOS machine, I always got a `bug: mkdir /homeless-shelter: permission denied` error with it, how did you install it?

**Comment by tomberek** [comment]
Yup, sandboxed builds. HOME is set to `/homeless-shelter` during build. 

When does the error happen?

**Comment by sondr3** [comment]
Hmm, I'm not sure what my `$HOME` is when I build, I'm installing with `nix-env -f . -i jump` in my `nixpkgs` directory, are you doing something other than that? And it happens when I try to run the [binary](https://github.com/NixOS/nixpkgs/pull/60631/files#diff-50b2060937e50c9c0360fd4f895132acR22), it tries to access the configuration directory that `jump` uses I think and I get the error. If I uncomment any of those lines it works fine.

File: pkgs/tools/system/jump/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  };
+
+  modSha256 = "1fzsm85c31vkdw80kijxmjhk8jyhjz8b21npgks2qrnizhm6iaf8";
+
+  outputs = [ "out" "man"];
+  postInstall = ''
+    install -D --mode=444 man/j.1 man/jump.1 -t $man/man/man1/
+
+    # generate completion scripts for jump
+    export HOME=$(mktemp -d)
```
Comment (kalbasit) [inline pkgs/tools/system/jump/default.nix]: If `HOME` is really required, please set it to $NIX_BUILD_TOP instead. See https://github.com/NixOS/nixpkgs/blob/0a1ed299f443f31bd5d0b849540002a5f15104c1/pkgs/build-support/build-bazel-package/default.nix#L22 for an example.

**Review by kalbasit — CHANGES_REQUESTED** [review]

File: pkgs/tools/system/jump/default.nix (hunk 2 above)
Comment (sondr3) [reply pkgs/tools/system/jump/default.nix]: Done, I cannot get it to work without `$HOME`, so I did what you suggested.

**Comment by kalbasit** [comment]
@GrahamcOfBorg build jump

**Review by kalbasit — APPROVED** [review]

**Pull request merged**